operator/pkg/clustermesh/endpointslice_meta_controller.go

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package clustermesh
+
+import (
+	"context"
+	"reflect"
+	"time"
+	"unsafe"
+
+	"github.com/cilium/endpointslice-controller/endpointslice"
+	"k8s.io/client-go/informers"
+
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/k8s/utils"
+)
+
+func newEndpointSliceMeshController(
+	meshConfig ClusterMeshConfig, meshPodInformer *meshPodInformer,
+	meshServiceInformer *meshServiceInformer, meshNodeInformer *meshNodeInformer,
+	clientset k8sClient.Clientset,
+) (*endpointslice.Controller, informers.SharedInformerFactory) {
+	meshClient := meshClient{clientset}
+
+	factory := informers.NewSharedInformerFactory(meshClient, 10*time.Minute)
+	endpointSliceInformer := factory.Discovery().V1().EndpointSlices()
+
+	controller := endpointslice.NewController(
+		context.Background(),
+		meshPodInformer,
+		meshServiceInformer,
+		meshNodeInformer,
+		endpointSliceInformer,
+		int32(meshConfig.ClusterMeshMaxEndpointsPerSlice),
+		meshClient, meshConfig.ClusterMeshEndpointUpdatesBatchPeriod,
```

Comment:
We should specify the controller name here with the `endpointslice.NewControllerWithName` function.